### PR TITLE
Added missing initialization, cleaned-up printing, removed dependencies

### PR DESCRIPTION
**Pull request by khues**
- Replaced direct call to "printf" by PRINTF macro
- Replaced direct call to "nbr_table_get_from_lladdr" by "uip_ds6_nbr_ll_lookup" to improve module encapsulation
- Added missing initialization of "neighborroutememb"
- Replaced direct call to "uip_debug_ipaddr_print" by PRINT6ADDR macro

### CODE REVIEW

File: core/net/ipv6/uip-ds6-route.c [hunk 1]
```diff
@@ -420,8 +421,7 @@ uip_ds6_route_rm(uip_ds6_route_t *route)
 
     if(neighbor_route == NULL) {
       PRINTF("uip_ds6_route_rm: neighbor_route was NULL for ");
-      uip_debug_ipaddr_print(&route->ipaddr);
-      PRINTF("\n");
+      PRINT6ADDR(&route->ipaddr);
```
Comment (nvt) [inline core/net/ipv6/uip-ds6-route.c]: The newline character was removed from the output.

File: core/net/rpl/rpl-dag.c [hunk 2]
```diff
@@ -92,10 +92,10 @@ rpl_print_neighbor_list()
     rpl_parent_t *p = nbr_table_head(rpl_parents);
     clock_time_t now = clock_time();
 
-    printf("RPL: rank %u dioint %u, %u nbr(s)\n", curr_rank, curr_dio_interval, uip_ds6_nbr_num());
+    PRINTF("RPL: rank %u dioint %u, %u nbr(s)\n", curr_rank, curr_dio_interval, uip_ds6_nbr_num());
```
Comment (nvt) [inline core/net/rpl/rpl-dag.c]: This long line should be split.

File: core/net/rpl/rpl-dag.c [hunk 3]
```diff
@@ -113,7 +113,7 @@ rpl_get_nbr(rpl_parent_t *parent)
   linkaddr_t *lladdr = NULL;
   lladdr = nbr_table_get_lladdr(rpl_parents, parent);
   if(lladdr != NULL) {
-    return nbr_table_get_from_lladdr(ds6_neighbors, lladdr);
+    return uip_ds6_nbr_ll_lookup((uip_lladdr_t*) lladdr);
```
Comment (nvt) [inline core/net/rpl/rpl-dag.c]: Code style fix: <code>(uip_lladdr_t *)lladdr</code>

**Comment by nvt** [comment]
I'm setting a time out label on this PR because the author has not responded since September 2015.

**Comment by simonduq** [comment]
I would add that we want `rpl_print_neighbor_list` to use `printf` rather than `PRINTF` because the point of the function is to print something unconditionally. The function itself is called only with `DEBUG` enabled.

**Comment by simonduq** [comment]
closing after timeout

**Pull request closed without merging**